File: Python/stack/stack.py
# Implementacao classica de uma Pilha

class Stack:
    def __init__(self, max_length):
        self.items = [0] * max_length
        self.max_length = max_length
        self.pointer = 0

    def is_empty(self):
        return self.pointer == 0
    
    def push(self,item):
        if self.pointer >+ self.max_length:
            raise IndexError("Stack full")
        
        self.items[self.pointer] = item
        self.pointer += 1

    def pop(self):
        if self.is_empty():
            raise IndexError("Empty stack.")
        
        self.pointer -= 1
        return self.items[self.pointer]
        
    def peek(self):
        if self.is_empty():
            raise IndexError("Empty Stack")
        
        return self.items[self.pointer - 1]
    
    def size(self):
        return self.pointer

File: Java/java-algorithms/src/Stack/Stack.java
package Stack;

// Implementacao classica de uma Pilha

public class Stack<T> {
    private final Object[] items;
    private final int maxLength;
    private int pointer;

    public Stack(int maxLength) {
        this.maxLength = maxLength;
        this.items = new Object[maxLength];
        this.pointer = 0;
    }

    public boolean isEmpty(){
        return pointer == 0;
    }

    public void push(T item){
        if(pointer >= maxLength){
            throw new IndexOutOfBoundsException("Stack full");
        }
        items[pointer++] = item;
    }

    public T pop(){
        if(isEmpty()){
            throw new IndexOutOfBoundsException("Stack is empty");
        }
        T item = (T) items[--pointer];
        items[pointer] = null;
        return item;
    }

    public T peek(){
        if(isEmpty()){
            throw new IndexOutOfBoundsException("Stack is empty");
        }
        return (T) items[pointer - 1];
    }

    public int size(){
        return pointer;
    }
}
